pkg/discover/discover.go

```diff
@@ -0,0 +1,170 @@
+package discover
+
+import (
+	"bufio"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/distributed-technologies/helm-overdrive-app-discover/pkg/logging"
+	"gopkg.in/yaml.v3"
+)
+
+// Structs for handling the information that comes from the `app.yaml`
+type App struct {
+	ArgocdAppNamespace            string
+	ArgocdAppSourceRepoUrl        string
+	ArgocdAppSourceTargetRevision string
+	ArgocdAppSourcePath           string
+	ApplicationFolder             string
+	CreateNamespace               bool   `yaml:"createNamespace"`
+	Name                          string `yaml:"name"`
+	Namespace                     string `yaml:"namespace"`
+	Project                       string `yaml:"project"`
+	Source                        Source `yaml:"source"`
+}
+
+type Source struct {
+	HelmRepo     string `yaml:"helm_repo"`
+	ChartName    string `yaml:"chart_name"`
+	ChartVersion string `yaml:"chart_version"`
+}
+
+// Unmarshals a .yaml file into the app struct
+func (app *App) GetValuesFromYamlFile(path string) error {
+	file, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	return yaml.Unmarshal(file, &app)
+}
+
+//
+func Discover(folder string) error {
+	logging.Debug("folder: %v\n", folder)
+
+	yamlFiles, err := GetFiles(folder)
+	if err != nil {
+		return err
+	}
+
+	for _, path := range yamlFiles {
+		logging.Debug("path: %s\n", path)
+
+		var app App
+
+		app.GetValuesFromYamlFile(path)
+		app.getArgoCDEnvs()
+
+		// Remove the base from the path
+		tmpString := strings.ReplaceAll(path, folder+"/", "")
+		tmpStringArray := strings.Split(tmpString, "/")
+
+		// We can then assume, that the last two entris in the array represents the `app.yaml` file and the application folder
+		// this gives us the app path without having to assume, that the name in the `app.yaml` file and the name of the folder it's in is not the same
+		tmpString = strings.Join(tmpStringArray[:len(tmpStringArray)-1], "/")
+
+		app.ApplicationFolder = tmpString
+
+		logging.Debug("App: %v\n", app)
+
+		app.GenArgoCDApp()
+	}
+
+	return nil
+}
+
+// Using the Golang template engine to generate the ArgoCD app
+func (app *App) GenArgoCDApp() error {
+	template, err := template.New("tempalte").Parse(getTempalte())
+	if err != nil {
+		return err
+	}
+
+	return template.Execute(os.Stdout, app)
+}
+
+// Trying to fetch ArgoCD environments
+func (app *App) getArgoCDEnvs() {
+	app.ArgocdAppNamespace = os.Getenv("ArgocdAppNamespace")
```

Comment:
I think I'm the one to blame for this one :)

The convention for environment variables are upper case snake case (AFAIK), ex:
```suggestion
	app.ArgocdAppNamespace = os.Getenv("ARGO_CD_APP_NAMESPACE")
```
This of course means the Go variable name and environment variable name don't match, but that is fine IMO, as it is two different conventions.

Reply:
Yes, this is a case of careless mass edit :P
Good catch :)